sdk/src/main/kotlin/io/rover/sdk/data/graphql/GraphQlApiService.kt

```diff
@@ -50,70 +49,6 @@ internal class GraphQlApiService(
-                        // HttpClient response mapping.
-                        httpResponse.responseCode < 300 -> false
-                        // 3xx redirects
-                        httpResponse.responseCode < 400 -> false
-                        // 4xx request errors (we don't want to retry these; onus is likely on
-                        // request creator).
-                        httpResponse.responseCode < 500 -> false
-                        // 5xx - any transient errors from the backend.
-                        else -> true
-                    }
-                )
-            }
-            is HttpClientResponse.Success -> {
-                try {
-                    val body = httpResponse.bufferedInputStream.use {
-                        it.reader(Charsets.UTF_8).readText()
-                    }
-
-                    log.v("RESPONSE BODY: $body")
-                    when (body) {
-                        "" -> ApiResult.Error(ApiError.EmptyResponseData(), false)
-                        else -> {
-                            try {
-                                ApiResult.Success(
-                                    httpRequest.decode(body)
-                                )
-                            } catch (e: APIException) {
-                                log.w("API error: $e")
-                                ApiResult.Error<TEntity>(
-                                    ApiError.InvalidResponseData(e.message ?: "API returned unknown error."),
-                                    // retry is not appropriate when we're getting a domain-level
-                                    // error from the GraphQL API.
-                                    false
-                                )
-                            } catch (e: JSONException) {
-                                // because the traceback information has some utility for diagnosing
-                                // JSON decode errors, even though we're treating them as soft
-                                // errors, throw the traceback onto the console:
-                                log.w("JSON decode problem details: $e, ${e.stackTrace.joinToString("\n")}")
-
-                                ApiResult.Error<TEntity>(
-                                    ApiError.InvalidResponseData(e.message ?: "API returned unknown error."),
-                                    // retry is not appropriate when we're getting a domain-level
-                                    // error from the GraphQL API.
-                                    false
-                                )
-                            }
-                        }
-                    }
-                } catch (exception: IOException) {
-                    ApiResult.Error<TEntity>(exception, true)
-                }
-            }
-        }
-
     /**
      * Performs the given [GraphQlRequest] when subscribed and yields the result to the subscriber.
```

Comment:
update docstring

Reply:
This is okay as it is I think as I don't think the function i've referred to her has changed